### PR TITLE
Menu: Progress bar

### PR DESCRIPTION
**Pull request by Develer**
**What is this feature?**

This PR adds an achievement progress bar to the top profile menu.
image

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/core/components/AppChrome/TopBar/TopNavBarMenuProgressBar.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        Level: <span>{level}</span>
+      </div>
+      <div className={styles.progressBar}>
+        <span
+          className={styles.progressBarFill}
+          style={{ width: `${(currentAchievement / totalAchievement) * 100}%` }}
+        ></span>
+      </div>
+      <span className={styles.progressBarCaption}>
+        {`${currentAchievement} of ${totalAchievement} achievements earned`}
```
Comment (nmarrs) [inline public/app/core/components/AppChrome/TopBar/TopNavBarMenuProgressBar.tsx:28]: Feedback - we can remove this for time being and just have progress bar be split up into 5 level segments, filled up to current level + % of the users progress in that current level tier

File: public/app/core/components/AppChrome/TopBar/TopNavBarMenuProgressBar.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  level: string;
+  totalAchievement: number;
+  currentAchievement: number;
+}
+
+export function TopNavBarMenuProgressBar({ level, totalAchievement, currentAchievement }: Props) {
+  const styles = useStyles2(getStyles);
+
+  return (
+    <div>
```
Comment (nmarrs) [inline public/app/core/components/AppChrome/TopBar/TopNavBarMenuProgressBar.tsx:17]: make this clickable to navigate to achievements profile page

**Review by nmarrs — APPROVED** [review]

**Comment by Develer** [comment]
image

File: public/app/core/components/AppChrome/TopBar/TopNavBarMenuProgressBar.tsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+      ? getProgress(
+          achievementsListByLevel?.filter((achievement) => achievement.completed).length!,
+          achievementsListByLevel.length
+        )
+      : 0;
+    setLevelProgress(progressByLevel);
+  }, [level, achievementsList]);
+
+  return (
+    <a href="/profile">
```
Comment (nmarrs) [inline public/app/core/components/AppChrome/TopBar/TopNavBarMenuProgressBar.tsx:44]:
```suggestion
    <a href="/profile/achievements">
```

**Pull request merged**